classes/class-wc-connect-api-client.php

```diff
@@ -530,6 +554,21 @@ protected function authorization_header() {
 			return $authorization;
 		}
 
+		protected function request_signature_wccom( $token_secret ) {
+			$method = strtoupper( $_SERVER['REQUEST_METHOD'] );
+			$data = array(
+				'host'        => 'woocommerce.com', // host URL.
+				'request_uri' => '/wp-json/helper/1.0/subscriptions', // endpoint URL.
```

Comment:
The request URI is hard-coded right now since we're only working with the /subscriptions/ endpoint, is there a way we can make this dynamic based on the endpoint being called? The trouble I had with that is there is no reference to an endpoint at this point.

Reply:
I think having the URI hard-coded is fine for now, but was wondering if you had any thoughts about changing the parameters for this function to include the method, URI, secret, and body? A couple notes on this function to see any thoughts you'd have on them:

- In getting the method this way, wouldn't it use the method that the request it is currently processing was requested with? I'm thinking this signature should be generated without any consideration for what the request to the Connect server is and only focused on what the eventual request to WCCOM will be.
- So with that in mind, I was thinking the $data['body'] shouldn't be set based on the Connect server request body, but rather should be passed in to the function explicitly. For example, we might eventually make a POST request to WCCOM with a body of `{ product_id: 1234 }` or something like that, so we wouldn't want the rest of the Connect server request body included in the signature.
- I was wondering if the host could be pulled from this? https://github.com/woocommerce/woocommerce/blob/6e5cc0c2f2e2651b722f3806fa7c9e624f5b9204/includes/admin/helper/class-wc-helper-api.php#L32-L34 This would make it a little easier to test WCCOM in different environments.

Reply:
Agreed on all of these points!

Now passing in the endpoint and method. 

>I'm thinking this signature should be generated without any consideration for what the request to the Connect server is and only focused on what the eventual request to WCCOM will be.

You're right; we care about eventually sending a GET request to WCCOM so let's pass 'GET' as the method rather than reading the request made to WCS since it's actually a POST. That was a bug for sure.

>So with that in mind, I was thinking the $data['body'] shouldn't be set based on the Connect server request body, but rather should be passed in to the function explicitly.

Also agreed, this is done now.

Also using WC_Helper_API::url( $endpoint ) to generate the request url. Should be much less brittle now. :)